### DISCUSSION TITLE
Two Problems: Offline paths and Unable to run job handler

### DISCUSSION THREAD

**Discussion by deiop** [immich-app/immich#5040] [General]
When I run the "generate thumbnails" job, it throws one exception: 

immich_microservices     | [Nest] 7  - 11/14/2023, 2:42:13 PM   ERROR [JobService] Unable to run job handler (thumbnailGeneration/generate-jpeg-thumbnail): Error: Input file contains unsupported image format
immich_microservices     | [Nest] 7  - 11/14/2023, 2:42:13 PM   ERROR [JobService] Error: Input file contains unsupported image format
immich_microservices     | [Nest] 7  - 11/14/2023, 2:42:13 PM   ERROR [JobService] Object:
immich_microservices     | {
immich_microservices     |   "id": "e5e93b29-634d-410c-b4de-6053ee5d133a"
immich_microservices     | }
immich_microservices     |

Where can I find that file? There is nothing in the upload folders, and: 
pics:/opt/immich# find . -name "e5e93b29-634d-410c-b4de-6053ee5d133*"
pics:/opt/immich#

**Second one:** 

Under "repair" there is one file under "offline paths"
/usr/src/app/upload/encoded-video/13ac1213-78b2-4657-bd3f-c9c863ab40ff/54/1e/541e7cce-c2ea-43ee-8956-7b091986b437-MP.mp4

as said, this file is not there anymore :-) - how can I fix that?

**Comment by ericberard** [comment]
I would also like to know how to repair the offline paths. I intend to user Immich as a mechanism to get photos off my iPhone. Once they're backed up, I'd like to move them into the permanent folder structure that I already use. Once there they no longer need to be in Immich. 

Once I get more familiar and comfortable with Immich I may rely on it more heavily but I still think it would be nice to be able to archive my photos off of Immich in some way.

**Comment by aisbergde** [comment]
I have the same idea to move assets between default and external libraries from outside. I could imagine to keep them in the default library, because the planned features for rotation and basic edit should be great and are planned only for the default library, as I understand.

Anyway at least deduplication between all libraries using external tools is required.

I don't understand, what prevents developers from including offline management also into the default library. IMHO, they could be handled the same way as for external libraries. There are flags in the assets table. Or do I miss something? Or is it just for safety, to prevent users from deletings to much by mistake?

**Comment by ericberard** [comment]
I did find this script but it failed when I ran it.
https://immich.app/docs/guides/remove-offline-files

This is the script I ran:
awk -F\" '/entityId/ {print $4}' orphans.json | while read line; do curl --location --request DELETE 'http://x.x.x.x:2283/api/asset' --header 'Content- Type: application/json' --header 'x-api-key: MyLongAPIKey' --data '{ "force": true, "ids": ["'"$line"'"]}';done

Here's the error message I got:
{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400}{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400}{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400}{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400}{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400}

Here's the output of awk -F\" '/entityId/ {print $4}' orphans.json:
d0696dfe-af8e-46b2-ba3f-bda571141c38
dd7169d0-7148-4b4e-bb73-d338d2148574
bd6c0341-25f6-4383-8392-5772f117b424
6997e4ad-0134-4f64-837f-1ab8f30af55c
80bb8410-23fe-44e4-b94a-19af11eba7a4

Those ID's match the 5 assets for which I'm trying to repair the paths.
Has anyone figured this out yet?

**Reply by rishid** [reply]
I have a few paths I deleted from my library and ended up in Offline Paths. I took a few IDs from orphan.json and successfully ran this curl command to remove the assets.
```
$ curl -L -X DELETE 'https://yourdomain.com/api/asset' -H 'Content-Type: application/json' -H 'x-api-key: foobar' -d '{
  "force": true,
  "ids": [
    "e21d7378-e16c-4e7b-89bb-32faeaca19a0"
  ]
}'
```

**Reply by queeup** [reply]
@ericberard, I solve this problem with the correct API key. You need an API key of the user which owns orphan files. There is no general API key.

**Reply by ericberard** [reply]
It's my user account that uploaded the files and it's in my user account where I created the API key. 
So I'm still at a loss. Thanks though for taking the time to reply. It's nice to know others are trying to help in resolving this with me.

**Reply by user222008** [reply]
Hello,
I made a python script following the official documentation: https://immich.app/docs/api/delete-assets

However, I get the same error as described above by [ericberard](https://github.com/ericberard) , tcpdump example:

```
DELETE /api/assets HTTP/1.1
Host: x.x.x.x:2283
User-Agent: python-requests/2.28.1
Accept-Encoding: gzip, deflate
Accept: application/json
Connection: keep-alive
x-api-key: foobar
Content-Length: 144

{"force": true, "ids": ["1e0d1778-a097-4ea5-9975-f6071e6465e3", "76385909-5fb9-4eb3-8b30-400b86262960", "21257010-3122-4d19-ae73-cecdb77a3dcf"]}HTTP/1.1 400 Bad Request
X-Powered-By: Express
x-immich-cid: jevmzcuf
Content-Type: application/json; charset=utf-8
Content-Length: 163
ETag: "a3xxxx8"
Date: Sun, 13 Apr 2025 17:57:07 GMT
Connection: keep-alive
Keep-Alive: timeout=5

{"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400,"correlationId":"jevmzcuf"}
```
The ownerID of the asset is correctly set as the API Key created for, the key is created having permissions set as {all}.
Tried using only one AssetID, the same result.

Any Ideas?

Thanks in advance!

**Reply by willsc123** [reply]
I'm currently trying to add assets to albums via the API and I have run into the same issue of **_"message":["each value in ids must be a UUID","ids should not be empty","ids must be an array"],"error":"Bad Request","statusCode":400_**.

I'm using Python and I have managed to reduce the error to only _**"message":["ids must be an array"]**_ by converting the ids string to uuid with the uuid python module (import uuid) and not using json.dumps(). Here is the snippet of code of that part:

```
payload = {
    "ids": [
        uuid.UUID("3fa85f64-5717-4562-b3fc-2c963f66afa6")
    ]
}
```

But I'm stuck with this as I cannot figure out how to satisfy this array problem. Has anyone any hint of how to deal with this?

**Reply by willsc123** [reply]
For anyone who might be interested, I found the issue that was giving me the errors. I wasn't including 'Content-Type': 'application/json' on the headers because I was reusing the variable from before. Once I put that on the code it worked as expected. I took the code from the API documentation (addAssetsToAlbum):

```
payload = json.dumps({
  "ids": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6 (my asset id instead)"
  ]
})
headers = {
  'Content-Type': 'application/json',
  'Accept': 'application/json',
  'x-api-key': 'MYAPIKEYHERE'
}
response = requests.request("PUT", url, headers=headers, data=payload)
```
Hope it help's someone
